components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go

```diff
@@ -51,7 +322,111 @@ func (ss *BatchCheckService) RunBastionCheck(check string, config models.Config,
 		resp[i].CheckType = check
 	}
 	resultChan <- resp
+}
+
+func (ss *BatchCheckService) getPortsToOpenForCheck(check string) map[string]map[string][]int {
+	return ss.getCheckInstance(check).GetPortsForMockServer()
+}
 
+func (ss *BatchCheckService) getDeploymentState(config models.Config) (string, error) {
+	if config.Hardware.AutomateNodeCount > 0 {
+		for index, ip := range config.Hardware.AutomateNodeIps {
+			result, err := ss.getStatusFromNode(ip)
+			if err != nil && index == len(config.Hardware.AutomateNodeIps)-1 {
+				errMsg := "received no response for status api from any automate nodes"
+				ss.log.Error(errMsg)
+				return "", errors.New(errMsg)
+			}
+			if result != "" {
+				return result, nil
+			}
+		}
+	}
+	return "", errors.New("automate nodes not present")
+}
+
+func (ss *BatchCheckService) getStatusFromNode(ip string) (string, error) {
+	url := fmt.Sprintf("http://%s:%s%s", ip, ss.port, constants.STATUS_API_PATH)
+	resp, err := ss.httpRequestClient.MakeRequest(http.MethodGet, url, nil)
+	if err != nil {
+		ss.log.Error("Error while calling status API from batch check service:", err)
+		return "", err
+	}
+
+	var statusApiResponse models.StatusApiResponse
+	err = json.NewDecoder(resp.Body).Decode(&statusApiResponse)
+	if err != nil {
+		ss.log.Error("Error while unmarshalling response of status API from batch check service:", err)
+		return "", err
+	}
+
+	if len(*statusApiResponse.Result.Services) == 0 && statusApiResponse.Result.Error != "" {
+		return constants.PRE_DEPLOY, nil
+	}
+
+	return constants.POST_DEPLOY, nil
+}
+
+func (ss *BatchCheckService) startMockServerOnHostAndPort(host, port string, startMockServerRequestBody models.StartMockServerRequestBody, respChan chan models.MockServerFromBatchServiceResponse) {
+	url := fmt.Sprintf("%s%s:%s%s", "http://", host, port, constants.START_MOCK_SERVER)
+	resp, err := ss.httpRequestClient.MakeRequest(http.MethodPost, url, startMockServerRequestBody)
+	if err != nil && resp == nil {
+		ss.log.Error("Error occurred while starting mock server", err.Error())
```

Comment:
```suggestion
		ss.log.Error("Error occurred while making request to start mock server: ", err.Error())
```
as same msg is shown for stop request for this scenario

Reply:
Done